File: java/728.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ans = new ArrayList<Integer>();
        for(int i = left;i <= right;i++){
            if(selfD(i) == true){
                ans.add(i);
            }
        }
        return ans;
    }
    
    public boolean selfD(int num){
        for(char c:String.valueOf(num).toCharArray()){
            if(c == '0' || num % (c - '0') > 0){
                return false;
            }
}
        return true;
    }
}

File: go/728.go
func selfDividingNumbers(left int, right int) []int {
	ans := make([]int, 0)
	for i := left; i <= right; i++ {
		if self(i) {
			ans = append(ans, i)
		}
	}
	return ans
}
func self(i int) bool {

	j := i
	for j > 0 {
		k := j % 10
		if k == 0 || i%k != 0 {
			return false
		}
		j /= 10
	}
	return true
}